src/sw/redis++/connection.cpp

```diff
@@ -167,42 +167,42 @@ auto ConnectionOptions::_split_uri(const std::string &uri) const
         throw Error("invalid URI: no scheme");
     }
 
-    auto type = uri.substr(0, pos);
+    auto scheme = uri.substr(0, pos);
 
     auto start = pos + 3;
     pos = uri.find("@", start);
     if (pos == std::string::npos) {
         // No auth info.
-        return std::make_tuple(type, std::string{}, uri.substr(start));
+        return std::make_tuple(scheme, std::string{}, uri.substr(start));
     }
 
     auto auth = uri.substr(start, pos - start);
 
-    return std::make_tuple(type, auth, uri.substr(pos + 1));
+    return std::make_tuple(scheme, auth, uri.substr(pos + 1));
 }
 
-auto ConnectionOptions::_split_path(const std::string &path) const
+auto ConnectionOptions::_split_path(const std::string &ipath) const
```

Comment:
Rename `ipath` as `spath` to be consistent with above changes. Please also rename other `ipath` variables.